### ISSUE TITLE
Barebones digital RGBHV input

### ISSUE THREAD

**Issue by jbshelton** [hoglet67/RGBtoHDMI#230]
I've been tinkering with the source code to this project in order to get it to work with very simple digital RGB signals meant for an LCD (specifically signals from the Gameboy, Gameboy Color, Gameboy Advance, etc.), without the use of a CPLD.
At first I had trouble properly configuring the vertical resolution because the firmware wouldn't allow anything below 250 lines per frame, but I modified geometry.c in order to accommodate for it. The OSD behaves kind of funky, but I can live with that.

One problem I need to solve is vsync: I need to have hsync, vsync, and psync on separate pins. All I could glean from the source code is that either vsync and hsync are XORed together in the CPLD for mixed/composite sync, or they are multiplexed onto the psync pin (though with wording for separate hsync and vsync, it's kind of confusing.) Ideally, there would already be an option for a separate vsync pin, but I couldn't find one; I wouldn't mind having to use an external XOR gate, but would like to know if I'm missing something in the source so I can just change the firmware.

I also am not completely sure how to configure the input signal resolution, based on exact measurements- I'm testing with the GBA at the moment, and I know the physical frame size is 308x228, and the visible size is 240x160, and the pixel clock frequency is 4194304Hz. I'm not sure if it's because currently don't have a way to feed vsync into the Pi, but the source detection does not want to abide by the parameters I put in.

Basically, I'm trying to add support for devices where the exact pixel clock frequency and resolution is known, and the output is in digital RGB with separate horizontal and vertical sync signals.

**Comment by IanSB** [comment]
When we were first experimenting with adding support for additional computers beyond the BBC micro, some support for separate H and V sync was added by XORing the two signals inside the CPLD and then feeding composite sync to the Pi. This worked OK as far as the sync signals themselves were concerned but it caused a problem because in order to support automatic detection and selection of sub-profiles for the various IBM PC modes, the polarity of the H & V sync pulses had to be determined and that info was lost when they were XORed together.

To work around this, the VSYNC input was fed to the PSYNC pin on the Pi when the VERSION output from the Pi was taken low (not enough spare pins for a direct connection). This means that the PSYNC signal is unavailable during VSYNC detection but it isn't required at that time so not a problem.

The references to mixed sync have been left in primarily to support old CPLD builds. Note the BBC CPLD doesn't support VSYNC at all so it gets detected as a mixed sync CPLD but never works with separate syncs.

If you have put the software into simple (non-CPLD) mode by tying a 3K3 resistor from the VERSION line (Pin 12 / GPIO 18) of the Pi zero header to ground (like the Amiga boards) then that only supports standard negative going composite sync so your best bet for now would be to xor the H and V signals and feed the resulting composite sync into the csync input (making sure it's the right polarity). You could always make custom changes later for separate vsync on another GPIO if you get it working.
